### PR TITLE
Calendar icon fix  #3525

### LINKED ISSUE #3525: Calender icon
datepicker comes with normal text field view, blank. Could there be an icon? which is kind of good for the UI, I guess.

### PR DESCRIPTION
**Pull request by febababy**
Calendar icon fix https://github.com/Hacker0x01/react-datepicker/issues/3525

Screenshot from 2023-02-01 11-28-29

### CODE REVIEW

**Comment by martijnrusschen** [comment]
I like the idea. To make this less disruptive, I'd like to have this as an optional prop that can be set. This makes it possible for users to add this icon if they want to, but aren't forced to use it.

**Comment by febababy** [comment]
@martijnrusschen Sure will make the icon an optional prop

File: docs-site/src/examples/default.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 () => {
   const [startDate, setStartDate] = useState(new Date());
   return (
-    <DatePicker selected={startDate} onChange={(date) => setStartDate(date)} />
+    <DatePicker
+      showIcon
```
Comment (martijnrusschen) [inline docs-site/src/examples/default.js:5]: Can you create a separate example in the docs site that shows how to make a datepicker with an icon in the input?

File: src/index.jsx [hunk 2]
```diff
@@ -1176,8 +1177,22 @@ export default class DatePicker extends React.Component {
   };
 
   renderInputContainer() {
+    const { showIcon } = this.props;
     return (
-      <div className="react-datepicker__input-container">
+      <div
```
Comment (martijnrusschen) [inline src/index.jsx:1182]: Can you add a test that checks if the icon (or CSS class) is shown when the boolean is set to true.

File: src/stylesheets/datepicker.scss [hunk 3]
```diff
@@ -517,6 +517,15 @@
   position: relative;
   display: inline-block;
   width: 100%;
+  .react-datepicker__calendar-icon {
+    position: absolute;
+    padding: 0.5rem;
```
Comment (martijnrusschen) [inline src/stylesheets/datepicker.scss:522]:
```suggestion

  .react-datepicker__calendar-icon {
    position: absolute;
    padding: 0.5rem;
```

File: src/stylesheets/datepicker.scss [hunk 4]
```diff
@@ -517,6 +517,15 @@
...
+  }
+}
+.react-datepicker__view-calendar-icon {
+  input {
```
Comment (martijnrusschen) [inline src/stylesheets/datepicker.scss:526]:
```suggestion
}

.react-datepicker__view-calendar-icon {
  input {
```

File: src/index.jsx (hunk 2 above)
Comment (febababy) [reply src/index.jsx:1182]: Sure

**Comment by febababy** [comment]
@martijnrusschen I have added an example and test cases.

**Comment by martijnrusschen** [comment]
This looks good, in the screenshot you added the icon is shown for both the default and the example you added. Is that intentional?

**Comment by febababy** [comment]
The icon is optional
image

**Comment by martijnrusschen** [comment]
thank you!

**Pull request merged**